### DISCUSSION TITLE
All tutorials are outdated

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [KadeArchive/Kade-Engine#2152] [Q&A]
How do I add a song all the tutorials on youtube are outdated
Please help

**Comment by Jayspency** [comment]
once you install kade engine, before you compile your game, export the songs you want to add into the song's folder in assets/songs. then compile your game and then go to debug menu. type in your song name and press reload audio, and boom

**Reply by THEkatinamicrowave** [reply]
Thanks I thought you had to change something in Playstate.hx

**Reply by NK-523** [reply]
a few things are missing here like how if you only have .ogg files with out .mp3 files it wont work, or how you have to make sure the song name in the .JSON files is the same as the song name in the freeplaylist.txt or you will get a null object reference when you try to hover over it, and how you need to have all 3 difficulty's or it won't show up in the freeplay list.

**Reply by SanicBTW** [reply]
The .mp3 files in the songs folder are only required if you want to export to html5

**Reply by THEkatinamicrowave** [reply]
ok

**Reply by NK-523** [reply]
when I try to add a song without them it says it cant find Inst or voices for the song.
